src/background/adblocker.js

```diff
@@ -496,7 +498,17 @@ if (__PLATFORM__ === 'firefox') {
 
         if (redirect !== undefined) {
           request.blocked = true;
-          result = { redirectUrl: redirect.dataUrl };
+          // There's a possibility that redirecting to file URL can expose
+          // extension existence.
+          if (details.type !== 'xmlhttprequest') {
+            result = {
+              redirectUrl: chrome.runtime.getURL(
+                REDIRECT_RESOURCE_PATH_PREFIX + redirect.filename,
```

Comment:
Leaving a note:

Adblocker library will emit stub filename from `@remusao/small` library. The location of those resources could be found on:
- https://github.com/remusao/mono/tree/master/packages/small/src
- https://github.com/ghostery/adblocker/blob/7a4986642a64df23c70bf7d254e1453a87d74c6f/packages/adblocker/src/resources.ts#L313-L316